src/main/java/de/bonndan/nivio/input/SourceReferencesResolver.java

```diff
@@ -2,37 +2,75 @@
 
 import de.bonndan.nivio.input.dto.LandscapeDescription;
 import de.bonndan.nivio.util.URLHelper;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.lang.NonNull;
 
 import java.net.URL;
+import java.util.Objects;
 
 
 /**
  * Resolves source references into collections of item descriptions.
- *
- *
  */
 public class SourceReferencesResolver {
 
     private final InputFormatHandlerFactory formatFactory;
     private final ProcessLog log;
+    private final ApplicationEventPublisher eventPublisher;
 
-    public SourceReferencesResolver(InputFormatHandlerFactory formatFactory, ProcessLog logger) {
-        this.formatFactory = formatFactory;
-        this.log = logger;
+    public SourceReferencesResolver(@NonNull final InputFormatHandlerFactory formatFactory,
+                                    @NonNull final ProcessLog logger,
+                                    @NonNull final ApplicationEventPublisher eventPublisher
+    ) {
+        this.formatFactory = Objects.requireNonNull(formatFactory);
+        this.log = Objects.requireNonNull(logger);
+        this.eventPublisher = Objects.requireNonNull(eventPublisher);
     }
 
     public void resolve(final LandscapeDescription landscapeDescription) {
 
         URL baseUrl = URLHelper.getParentPath(landscapeDescription.getSource()).orElse(null);
         landscapeDescription.getSourceReferences().forEach(ref -> {
+            InputFormatHandler factory;
+            try {
+                factory = formatFactory.getInputFormatHandler(ref);
+            } catch (ProcessingException ex) {
+                log.error(ex.getMessage());
+                eventPublisher.publishEvent(new ProcessingErrorEvent(landscapeDescription, ex));
+                landscapeDescription.setIsPartial(true);
+                return;
+            } catch (RuntimeException ex) {
+                String msg = "Failed to resolve source reference '" + ref.getUrl() + "': " + ex.getMessage();
+                log.warn(msg);
+                landscapeDescription.setIsPartial(true);
+                return;
+            }
+
             try {
-                InputFormatHandler factory = formatFactory.getInputFormatHandler(ref);
                 landscapeDescription.addItems(factory.getDescriptions(ref, baseUrl));
             } catch (ProcessingException ex) {
-                log.warn("Failed to resolve source reference '" + ref.getUrl() + "' (" + ref.getFormat() + ")", ex);
+                String message = ex.getMessage();
+                if (ex instanceof ReadingException) {
+                    message += ": " + getCauseMessage(ex.getCause());
+                }
+                log.error(message);
+                eventPublisher.publishEvent(new ProcessingErrorEvent(landscapeDescription, ex));
+                landscapeDescription.setIsPartial(true);
+            } catch (RuntimeException ex) {
+                log.warn(ex.getMessage());
```

Comment:
This is basically just to have mor detailed error messages?

Reply:
Yes. We are forwarded JSON parse errors to the frontend.